src/components/TodoContentMain.tsx

```diff
@@ -0,0 +1,213 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import {
+  FC, useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { todosApi } from '../api/todos-api';
+import { useTodoContext } from '../context/todoContext/useTodoContext';
+import { useErrorContext } from '../context/errorContext/useErrorContext';
+import { UpdateTodoArgs } from '../types/UpdateTodoArgs';
+
+interface TodoContentMainProps {
+  todos: Todo[],
+  tempTodo: Todo | null,
+}
+
+export const TodoContentMain: FC<TodoContentMainProps> = (props) => {
+  const { todos, tempTodo } = props;
+  const {
+    removeTodos,
+    setHandlingTodoIds,
+    handlingTodoIds,
+    updateTodos,
+  } = useTodoContext();
+  const { notifyAboutError } = useErrorContext();
+  const [editedTodoId, setEditedTodoId] = useState<number | null>(null);
+  const todoTitleRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    todoTitleRef.current?.focus();
+  }, [editedTodoId]);
+
+  const onRemove = async (id: number) => {
+    setHandlingTodoIds([id]);
+    try {
+      const result = await todosApi.remove([id]);
+
+      if (result) {
+        removeTodos([id]);
+      }
+    } catch {
+      notifyAboutError('Unable to delete a todo');
+    } finally {
+      setHandlingTodoIds([]);
+    }
+  };
+
+  const onEdit = async (id: number, data: UpdateTodoArgs) => {
+    try {
+      setHandlingTodoIds([id]);
+      const updatedTodos = await todosApi.update([{
+        id,
+        data,
+      }]);
+
+      updateTodos(updatedTodos);
+    } catch {
```

Comment:
In the `catch` block, you should also catch a possible `error`, because now the error does not look informative, for example, you can do this:
```
catch(error) { notifyAboutError(`Unable to update a todo: ${error.message}`)}
```